packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -29,13 +29,13 @@ import { ISharedObject, ISharedObjectEvents } from "./types";
 /**
  *  Base class from which all shared objects derive
  */
-export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedObjectEvents>
+export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISharedObjectEvents>
     extends EventEmitterWithErrorHandling<TEvent> implements ISharedObject<TEvent> {
     /**
      * @param obj - The thing to check if it is a SharedObject
      * @returns Returns true if the thing is a SharedObject
      */
-    public static is(obj: any): obj is SharedObject {
+    public static is(obj: any): obj is SharedObjectCore {
```

Comment:
It's not directly related to your changes, but I believe this is wrong - opened https://github.com/microsoft/FluidFramework/issues/8594 to track it